File: Midterm_example/src/main/java/pullUpMethod/After.java
package pullUpMethod;

class Employee{
    /**
     *- Putting duplicated method to superclass Employee
     *- Delete duplicated method on subclas Saleman and Engineer
     */
    public String getName(){
        /** ... */
        return "";
    }
}

class Saleman extends Employee{
}

//File Engineer.java
class Engineer extends Employee{
}

File: CSharpGenelization 3/CSharpGenelization/PullUpMethod/After.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpGenelization.PullUpMethod
{
    internal class After
    {

        //Putting duplicated method to superclass Employee
        //Delete duplicated method on subclas Saleman and Engineer
        class Employee
        {
            protected String getName()
            {
                return "";
            }
        }

        class Saleman : Employee { }

        class Engineer : Employee { }
    }
}
